MinimedKitTests/TimestampedHistoryEventTests.swift

```diff
@@ -0,0 +1,92 @@
+    var historyPage: HistoryPage!
+    var historyPage522: HistoryPage!
+
+    
+    override func setUp() {
+        super.setUp()
+        do {
+            historyPage = try HistoryPage(pageData: hexData, pumpModel: PumpModel.Model523)
+            historyPage522 = try HistoryPage(pageData: hexData522, pumpModel: PumpModel.Model522)
+        } catch {
+            
+        }
+    }
+    
+    func testEventIsntMutable() {
+        let data = Data(bytes: Array<UInt8>([7,6,5,4,3,2,1,0]))
+        let event = BatteryPumpEvent(availableData: data, pumpModel: PumpModel.Model523)!
+        
+        let sut = TimestampedHistoryEvent(pumpEvent:event, date:Date())
+        
+        XCTAssertFalse(sut.isMutable())
+    }
+    
+    func testEventIsNotMutableFor522() {
+        let data = Data(bytes: Array<UInt8>([7,6,5,4,3,2,1,0]))
+        let event = BatteryPumpEvent(availableData: data, pumpModel: PumpModel.Model522)!
+        
+        let sut = TimestampedHistoryEvent(pumpEvent:event, date:Date())
+        
+        XCTAssertFalse(sut.isMutable())
+    }
+
+    func test523EventIsNotMutable() {
+        
+        let bolusEvent = getNormalBolusEvent()
+       
+        let timeStampDate = bolusEvent.timestamp.date!
+        let dateToCheck = timeStampDate.addingTimeInterval(bolusEvent.deliveryTime + 10) // beyond the delivery time
+        
+        let sut = TimestampedHistoryEvent(pumpEvent: bolusEvent, date: timeStampDate)
+        
+        XCTAssertFalse(sut.isMutable(atDate: dateToCheck))
+    }
+    
+    func test523EventIsMutable() {
+        
+        let bolusEvent = getNormalBolusEvent()
+        
+        let timeStampDate = bolusEvent.timestamp.date!
+        let dateToCheck = timeStampDate.addingTimeInterval(bolusEvent.deliveryTime/2) // within the delivery time
+        
+        let sut = TimestampedHistoryEvent(pumpEvent: bolusEvent, date: timeStampDate)
+        
+        XCTAssertTrue(sut.isMutable(atDate: dateToCheck))
+    }
+
+// Not sure if this event is actually mutable - need to verify per ps2's comment on PR
+//    func test522EventIsMutable() {
+//        
+//        let tempBasalEvent = historyPage522.events[1] as! TempBasalPumpEvent
+//    
+//        var date = tempBasalEvent.timestamp.date!
+//        date.addTimeInterval(15)
+//        
+//        let sut = TimestampedHistoryEvent(pumpEvent: tempBasalEvent, date: date)
+//        
+//        XCTAssertTrue(sut.isMutable())
+//    }
+    
+    func getNormalBolusEvent() -> BolusNormalPumpEvent {
+        let events = historyPage.events
+        let bolus = events[1] as! BolusNormalPumpEvent
+        return bolus
+    }
+}
```

Comment:
I think the tests we want here are to assert we detect which events can appear out of order in history, and that the the fetch history command doesn't get tripped up by them when they are.